src/DictVectors/initiators.jl

```diff
@@ -0,0 +1,219 @@
+"""
+struct InitiatorValue{V}
+    safe::V
+    unsafe::V
+    initiator::V
+end
+function InitiatorValue{V}(;safe=zero(V), unsafe=zero(V), initiator=zero(V)) where {V}
+    return InitiatorValue{V}(V(safe), V(unsafe), V(initiator))
+end
+function InitiatorValue{V}(i::InitiatorValue) where {V}
+    return InitiatorValue{V}(V(i.safe), V(i.unsafe), V(i.initiator))
+end
+
+function Base.:+(v::InitiatorValue, w::InitiatorValue)
+    return InitiatorValue(v.safe + w.safe, v.unsafe + w.unsafe, v.initiator + w.initiator)
+end
+function Base.:-(v::InitiatorValue, w::InitiatorValue)
+    return InitiatorValue(v.safe - w.safe, v.unsafe - w.unsafe, v.initiator - w.initiator)
+end
+function Base.:-(v::InitiatorValue{V}) where {V}
+    return InitiatorValue{V}(-v.safe, -v.unsafe, -v.initiator)
+end
+Base.zero(::Union{I,Type{I}}) where {V,I<:InitiatorValue{V}} = InitiatorValue{V}()
+
+Base.convert(::Type{<:InitiatorValue{V}}, x::InitiatorValue{V}) where {V} = x
+function Base.convert(::Type{<:InitiatorValue{U}}, x::InitiatorValue{V}) where {U,V}
+    return InitiatorValue{U}(safe=x.safe, unsafe=x.unsafe, initiator=x.initiator)
+end
+function Base.convert(::Type{<:InitiatorValue{V}}, x) where {V}
+    return InitiatorValue{V}(safe=convert(V, x))
+end
+
+"""
+    AbstractInitiator
+
+Abstract type for defining initiator rules. Concrete implementations:
+
+* [`Initiator`](@ref)
+* [`SimpleInitiator`](@ref)
+* [`CoherentInitiator`](@ref)
+"""
+abstract type AbstractInitiator{V} end
+
+"""
+    Initiator
+
+Initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators can spawn to initiators.
+"""
+struct Initiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::Initiator, v::InitiatorValue)
+    return v.safe + v.initiator + !iszero(v.initiator) * v.unsafe
+end
+
+"""
+    SimpleInitiator
+
+Simplified initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators cannot spawn.
+"""
+struct SimpleInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::SimpleInitiator, v::InitiatorValue)
+    return v.safe + v.initiator
+end
+
+"""
+    CoherentInitiator
+
+Initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators can spawn to initiators.
+* Multiple non-initiators can spawn to a single non-initiator if their contributions add up
+  to a value greater than the initiator threshold.
+"""
+struct CoherentInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::CoherentInitiator, v::InitiatorValue)
+    if !iszero(v.initiator) || abs(v.unsafe) > i.threshold
+        return v.initiator + v.safe + v.unsafe
+    else
+        return v.initiator + v.safe
+    end
+end
+
+"""
+    InitiatorDVec
+
+Functionally identical to [`DVec`](@ref), but contains [`InitiatorValue`]s
+"""
```

Comment:
Instructions for how to construct such an object with kwargs should be added here.